tests/functional/docs/test_s3.py

```diff
@@ -27,11 +27,21 @@ def setUp(self):
         self.generated_contents = self.generated_contents.decode('utf-8')
 
     def test_file_transfer_methods_are_documented(self):
+        self.assert_contains_lines_in_order(
+            [
+                '=========',
+                'Resources',
+                '=========',
+                'The available resources are:',
+                '  s3/service-resource/index',
+            ],
+            self.generated_contents,
+        )
         self.assert_contains_lines_in_order(
             [
                 '.. py:class:: S3.Client',
-                '  s3/Client/download_file',
```

Comment:
Some of the naming changes are losing the casing that distinguishes classes from methods. Was that change prompted by a technical requirement or just to enforce path consistency?

Reply:
The lowercasing of sub-paths was prompted by the fact that resources are now in their own sub-pages, the fragments of section headers are lowercased, and CloudFront path patterns are case-sensitive. If a user was previously linking to the following resource section headers `/s3.html#service-resource` or `.../s3.html#bucketlifecycleconfiguration`, we would (without the lowercasing) want to reroute to `/s3/ServiceResource/index.html` which is mappable but we'd have to do the same for `/s3/BucketLifeCycleConfiguration/index.html` and all other resources (93 across 10 services). These are the options I see:
1. Leave as is and sacrifice casing.
2. Create a huge mapping of lowercased resource names to correctly cased names.
3. Don't support service resource section header redirections. Note: Resource class/method links will still be rerouted as normal since their fragments provide the correct casing.

Reply:
I think that's probably ok, as long as you'll reach the same location with the old link, I don't think casing is the end of the world.